src/components/tooltip/style.module.css

```diff
@@ -0,0 +1,96 @@
+/*
+Additional composition is helpful here.
+Note that .arrow and .root can't use CSS custom properties
+nicely since .arrow is rendered into a Portal.
+
+This also means that, unlike in other /partials,
+we can't reliably pull CSS custom properties from a
+theme.module.css class at the component root.
+*/
+
+.theme-light {
+  --background: var(--token-color-surface-primary);
+  --foreground: var(--token-color-foreground-strong);
+}
+
+.theme-dark {
+  --background: var(--token-color-foreground-strong);
+  --foreground: var(--token-color-surface-primary);
+}
+
+.arrowContainer {
+  /**
+   * Note: --top, --left, --offset, & --size are set in JS,
+   * to allow for dynamic, collision-free positioning.
+   *
+   * --top & --left position this element center-top aligned
+   * with the tooltip container.
+   * - We use translateY(-50%) to make half this element
+   *   protrude above the tooltip.
+   * - The .arrow element provides the visible tooltip arrow element,
+   *   which includes a box shadow that we need to clip, as otherwise it
+   *   would overlap on the tooltip
+   * - The clip-path clips the .arrow element to prevent it from
+   *   casting a box-shadow on the tooltip below. It also contains a -2px
+   *   clip inset at the top, to ensure we don't clip off the top corner
+   *   of the .arrow element, which protrudes a bit.
+   */
+  clip-path: inset(-2px 0 50% 0);
+  height: var(--size);
+  left: var(--left);
+  position: absolute;
+  transform: translateY(-48%); /* not 50% avoids off-by-a-bit rounding issues */
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  top: var(--top);
+  width: var(--size);
+  z-index: 2;
+}
+
+.arrow {
+  /**
+   * We want to set the width of this .arrow relative to the parent width,
```

Comment:
Very much leaning into really descriptive comments here... otherwise some of the stuff here would probably look like magic numbers 🤷‍♂️